web/src/pages/api/public/v2/prompts/[promptName].ts

```diff
@@ -0,0 +1,88 @@
+import { verifyAuthHeaderAndReturnScope } from "@/src/features/public-api/server/apiAuth";
+import { cors, runMiddleware } from "@/src/features/public-api/server/cors";
+import { isPrismaException } from "@/src/utils/exceptions";
+import { prisma } from "@langfuse/shared/src/db";
+import { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+const GetPromptSchema = z.object({
+  promptName: z.string(),
+  version: z.coerce.number().optional(),
+  active: z.coerce.boolean().optional(),
+});
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  await runMiddleware(req, res, cors);
+  if (req.method !== "GET") {
+    console.error(req.method, req.body, req.query);
+    return res.status(405).json({ message: "Method not allowed" });
+  }
+
+  try {
+    const authCheck = await verifyAuthHeaderAndReturnScope(
+      req.headers.authorization,
+    );
+    if (!authCheck.validKey)
+      return res.status(401).json({
+        message: authCheck.error,
+      });
+
+    if (authCheck.scope.accessLevel !== "all") {
+      return res.status(401).json({
+        message:
+          "Access denied - need to use basic auth with secret key to GET traces",
+      });
+    }
+
+    console.log("Trying to get prompt:", req.body, req.query);
+
+    const { promptName, version, active } = GetPromptSchema.parse(req.query);
```

Comment:
uridecoding promptName makes sense to be able to parse special characters in the name

Reply:
added